mesonbuild/modules/unstable_cuda.py

```diff
@@ -0,0 +1,229 @@
+        if not isinstance(detected, (str, list)):
+            raise InvalidArguments('"detected" keyword must be a string or list of strings.')
+        ret = self._get_nvcc_arch_flags(cuda_version, args, detected)[1]
+        return ModuleReturnValue(ret, [ret])
+
+    def _get_nvcc_arch_flags(self, cuda_version, cuda_arch_list="Auto", detected=""):
+        """
+        Using the CUDA version (the NVCC version) and the target architectures,
+        compute the nvcc architecture flags.
+        """
+
+        cuda_known_gpu_architectures  = ["Fermi", "Kepler", "Maxwell"]  # noqa: E221
+        cuda_common_gpu_architectures = ["3.0", "3.5", "5.0"]           # noqa: E221
+        cuda_limit_gpu_architecture   = None                            # noqa: E221
+        cuda_all_gpu_architectures    = ["3.0", "3.2", "3.5", "5.0"]    # noqa: E221
+
+        if version_compare(cuda_version, "<7.0"):
+            cuda_limit_gpu_architecture = "5.2"
+
+        if version_compare(cuda_version, ">=7.0"):
+            cuda_known_gpu_architectures  += ["Kepler+Tegra", "Kepler+Tesla", "Maxwell+Tegra"]  # noqa: E221
+            cuda_common_gpu_architectures += ["5.2"]                                            # noqa: E221
+
+            if version_compare(cuda_version, "<8.0"):
+                cuda_common_gpu_architectures += ["5.2+PTX"]  # noqa: E221
+                cuda_limit_gpu_architecture    = "6.0"        # noqa: E221
+
+        if version_compare(cuda_version, ">=8.0"):
+            cuda_known_gpu_architectures  += ["Pascal"]             # noqa: E221
+            cuda_common_gpu_architectures += ["6.0", "6.1"]         # noqa: E221
+            cuda_all_gpu_architectures    += ["6.0", "6.1", "6.2"]  # noqa: E221
+
+            if version_compare(cuda_version, "<9.0"):
+                cuda_common_gpu_architectures += ["6.1+PTX"]  # noqa: E221
+                cuda_limit_gpu_architecture    = "7.0"        # noqa: E221
+
+        if version_compare(cuda_version, ">=9.0"):
+            cuda_known_gpu_architectures  += ["Volta"]                             # noqa: E221
+            cuda_common_gpu_architectures += ["7.0", "7.0+PTX"]                    # noqa: E221
+            cuda_all_gpu_architectures    += ["7.0", "7.0+PTX", "7.2", "7.2+PTX"]  # noqa: E221
+
+            if version_compare(cuda_version, "<10.0"):
+                cuda_limit_gpu_architecture = "7.5"
+
+        if version_compare(cuda_version, ">=10.0"):
+            cuda_known_gpu_architectures  += ["Turing"]          # noqa: E221
+            cuda_common_gpu_architectures += ["7.5", "7.5+PTX"]  # noqa: E221
+            cuda_all_gpu_architectures    += ["7.5", "7.5+PTX"]  # noqa: E221
+
+            if version_compare(cuda_version, "<11.0"):
+                cuda_limit_gpu_architecture = "8.0"
+
+        if not cuda_arch_list:
+            cuda_arch_list = "Auto"
+
+        if   cuda_arch_list == "All":     # noqa: E271
```

Comment:
I don't see any advantage in ignoring E271 on three lines just so you can align this one line with the others.

Reply:
My personal preference would have been to disable `E221`, `E271` or even Flake8 entirely, so that I wouldn't have to put such comments on one or three or N lines and could freely align code for aesthetic or readability purposes.

But, be that as it may, Flake8 and those warnings are all enabled. Putting `# noqa: E271` on just that line would make one wonder what makes it so special. Putting it on all three lines makes none of the lines stand out.

This type of redundancy is present in many other lines as well.

Reply:
Yes, well, your preferences aside, it _is_ enabled, and I don't see any point in making exceptions just to align this one `if` statement.

Reply:
> Yes, well, your preferences aside, it _is_ enabled,

I'm quite well aware of that, which is why I did add `# noqa: E271`.

> and I don't see any point in making exceptions just to align this one `if` statement.

I made a point above about the lines with suppressions standing out (like that `if`), unless equally-important lines also have them redundantly (like the `elif`'s).

But it is still unclear to me what actions you'd like me to take.

- If you insist on me removing all **redundant** suppressions, I'll remove them, even if it thereby creates a "signal" for the reader that isn't really there.
- If you insist on me removing **all** suppressions and not dealigning, I'll remove them, but it will lead to the automated Flake8 check failing.
- If you insist on me removing **all** suppressions and dealigning, I'll remove them and dealign, but I (and many others as well) will consider the code less readable thenceforth.

We only have to decide which of these resolutions the Meson project would prefer.